packages/studio-plugin/src/createStudioPlugin.ts

```diff
@@ -33,6 +35,24 @@ export default async function createStudioPlugin(
   const pathToUserProjectRoot = getProjectRoot(cliArgs);
 
   const studioConfig = await getStudioConfig(pathToUserProjectRoot, cliArgs);
+
+  let pagesDevelopmentPort: Promise<number | undefined>;
+  if (studioConfig.isPagesJSRepo) {
+    pagesDevelopmentPort = startPagesDevelopmentServer();
+    pagesDevelopmentPort
+      .then((port) => {
+        if (port) {
+          process.on("exit", () => {
+            console.log("Studio is shutting down");
+            execSync(`npx kill-port ${port}`);
+          });
+        }
+      })
```

Comment:
I think I would prefer for this `.then(() => ...` logic to be inside of startPagesDevelopmentServer?